File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        dp = [0] * (n + 1)
        dp[1] = cost[0]
        for i in range(2, n):
            dp[i] = cost[i-1] + min(dp[i - 1], dp[i - 2])
            # print(i, dp, cost[i-1], dp[i - 1], dp[i - 2])
        dp[n] = min(dp[n-1], dp[n-2]+cost[n-1])
        return dp[n]

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.go
// import "fmt"

func minCostClimbingStairs(cost []int) int {
	n := len(cost)
	dp := make([]int, n+1)
    dp[1] = cost[0]
	for i := 2; i < n; i++ {
        dp[i] = cost[i-1] + min(dp[i-2], dp[i-1])
		// fmt.Println(i, dp[i-1], cost[i-1], cost[i], dp)
	}
    dp[n] = min(dp[n-2]+cost[n-1], dp[n-1])
	// fmt.Println(dp)
	return dp[n]
}